backend/src/upload/upload.ts

```diff
@@ -147,13 +151,21 @@ export function find_surveys(
       }
     });
 
+    const firstXML = index_xml_files_in_HDC_directory(
+      surveyDirectory + '/HDC',
+    )[0];
+    const dynatestId = get_surveyID(firstXML, debug);
+    const surveyDate = get_date(firstXML, debug);
```

Comment:
```suggestion
    const xmls = index_xml_files_in_HDC_directory(
      surveyDirectory + '/HDC',
    );
    const dynatestId = get_surveyID(xmls[0], debug);
    const surveyDate = get_date(xmls[0], debug);
```
to avoid recomputing `index_xml_files_in_HDC_directory`